cfgov/v1/models/molecules.py

```diff
@@ -117,16 +117,77 @@ class TextIntroduction(blocks.StructBlock):
 
     class Meta:
         icon = 'title'
-        template = 'v1/demo/molecules/text_introduction.html'
+        template = '_includes/molecules/text-introduction.html'
+
+
+class Hero(blocks.StructBlock):
+    heading = blocks.CharBlock(max_length=100, required=True)
+    body = blocks.RichTextBlock(required=False)
+
+    image = ImageChooserBlock(required=False)
+    image_path = blocks.CharBlock(required=False)
+    image_alt = blocks.CharBlock(required=False)
+
+    background_color = blocks.CharBlock(max_length=100, required=False)
+    link_url = blocks.URLBlock(required=False)
+    link_text = blocks.CharBlock(max_length=100, required=False)
+    is_button = blocks.BooleanBlock(required=False)
+
+    def clean(self, data):
+        error_dict = {}
+        try:
+            block_data = super(Hero, self).clean(data)
+        except ValidationError as e:
+            error_dict.update(e.params)
+            block_data = data
+
+        if not block_data['image'] and not block_data['image_path'] and not block_data['image_alt']:
+            img_err = ['Please upload or enter an image path']
+            error_dict.update({'image': img_err, 'image_path': img_err, 'image_alt': isRequired('Image alt')})
+
+        if block_data['image'] and block_data['image_path']:
+            img_err = ['Please select one method of image rendering']
+            error_dict.update({
+                'image': img_err,
+                'image_path': img_err})
+
+        if block_data['image_path'] and not block_data['image_alt']:
+            error_dict.update({'image_alt': isRequired('Image Alt')})
+
+        if error_dict:
+            raise ValidationError("Hero validation errors", params=error_dict)
+        else:
+            return block_data
+
+    class Meta:
+        icon = 'image'
+        template = '_includes/molecules/hero.html'
+
+
+class FormFieldWithButton(blocks.StructBlock):
+    btn_text = blocks.CharBlock(max_length=100, required=True)
+
+    required = blocks.BooleanBlock(required=False)
+    id = blocks.CharBlock(max_length=100, required=False)
+    info = blocks.RichTextBlock(required=False)
+    label = blocks.CharBlock(max_length=100, required=True)
+    type = blocks.CharBlock(max_length=100, required=True)
```

Comment:
Now I'm wondering if this, and other fields, can be passed in when the block is declared.

For example from the Email org: `form_field = blocks.ListBlock(molecules.FormFieldWithButton(type='email'), icon='mail', required=False)`